client/client.go

```diff
@@ -1270,6 +1272,13 @@ func (c *Client) restoreState() error {
 			continue
 		}
 
+		allocState, err := c.stateDB.GetAcknowledgedState(alloc.ID)
+		if err != nil {
+			c.logger.Error("error restoring last acknowledged alloc state, will update again",
+				err, "alloc_id", alloc.ID)
+		}
+		ar.AcknowledgeState(allocState)
```

Comment:
It'll still be nil in this case, so probably better just to put the `ar.AcknowledgeState` in the `else` block for that error so that if we change `AcknowledgeState` later it's that much harder to introduce a NPE.

Reply:
Resolved in f62c33f7e4fc40db35ed3287d8089e6258a3fe58